cms/djangoapps/coursegraph/tasks.py

```diff
@@ -388,36 +388,31 @@ def dump_courses_to_neo4j(self, credentials, override_cache=False):
                 reason = "override_cache is True"
             log.info("submitting %s, because %s", course_key, reason)
             dump_course_to_neo4j.apply_async(
-                args=[str(course_key), credentials],
+                kwargs=dict(
+                    course_key_string=str(course_key),
+                    connection_overrides=connection_overrides,
+                )
             )
             submitted_courses.append(str(course_key))
 
         return submitted_courses, skipped_courses
 
 
-def authenticate_and_create_graph(credentials):
+def authenticate_and_create_graph(connection_overrides=None):
     """
     This function authenticates with neo4j and creates a py2neo graph object
+
     Arguments:
-        credentials (dict): a dictionary of credentials used to authenticate,
-          and then create, a py2neo graph object.
+        connection_overrides (dict): overrides to Neo4j connection
+            parameters specified in `settings.COURSEGRAPH_CONNECTION`.
 
     Returns: a py2neo `Graph` object.
     """
-
-    host = credentials['host']
-    port = credentials['port']
-    secure = credentials['secure']
-    neo4j_user = credentials['user']
-    neo4j_password = credentials['password']
-
-    graph = Graph(
-        protocol='bolt',
-        password=neo4j_password,
-        user=neo4j_user,
-        address=host,
-        port=port,
-        secure=secure,
-    )
-
-    return graph
+    connection_args = {}
+    for key in ['protocol', 'host', 'port', 'secure', 'user', 'password']:
+        default = settings.COURSEGRAPH_CONNECTION.get(key)
+        override = (connection_overrides or {}).get(key)
+        # Note: We intentionally did not write this as "... = override or default"
+        # because `False` is a legimitate override value for 'secure'.
+        connection_args[key] = override if override is not None else default
```

Comment:
Nit (completely optional): This is totally fine as-is, but using a dict's `update` method can make what's happening a little easier to follow, e.g.:

```python
connection_args = {
    key: settings.COURSEGRAPH_CONNECTION.get(key)
    for key in ['protocol', 'host', 'port', 'secure', 'user', 'password']
}
if connection_overrides:
    connection_args.update(connection_overrides)
```

Reply:
I fiddled with this a bit and landed on:
```python
provided_overrides = {
    key: value
    for key, value in connection_overrides.items()
    # Drop overrides whose values are `None`. Note that `False` is a
    # legitimate override value that we don't want to drop here.
    if value is not None
}
connection_with_overrrides = {**settings.COURSEGRAPH_CONNECTION, **provided_overrides}
return Graph(connection_with_overrrides)
```
(On reflection, fitlering on the list of valid keys seemed like a bad idea since it'd silently drop any typo'd connection keys instead of letting py2neo scream about them naturally)

Reply:
Related change: https://github.com/openedx/edx-platform/pull/29156/files#r834900825